packages/devtools/test/support/flutter_test_environment.dart

```diff
@@ -79,7 +80,7 @@ class FlutterTestEnvironment {
   }
 
   Future<void> tearDownEnvironment({bool force = false}) async {
-    if (!force && reuseTestEnvironment) {
+    if (!_needsTeardown || (!force && reuseTestEnvironment)) {
```

Comment:
nit: it would be simpler to remove the _needsTeardown getter and make this if
if (_needsSetup || (!force && reuseTestEnvironment)) {
because positive conditions are easier to read.

Reply:
I thought it might be less obvious why needing setup means we don't need to teardown but I think you're right this does look a little hard to digest now. I'll tweak.